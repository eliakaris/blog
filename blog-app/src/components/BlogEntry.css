article
{
    width: 720px;
}

article img
{
    display: block;
    padding: 5px;
    max-width: 720px
}