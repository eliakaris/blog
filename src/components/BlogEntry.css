article
{
    width: 720px;
    margin: auto;
}

article img
{
    display: block;
    padding: 5px;
    max-width: 720px;
    border-radius: 4px;
    box-shadow: 5px 5px 5px #ccc;
}