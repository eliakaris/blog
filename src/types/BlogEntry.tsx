// src/types/BlogEntry.tsx

export interface BlogEntryData {
  pretty_pub_date: string;
  html: string;
  summary: string;
  slug: string;
  title: string;
  pub_date: string;
}

export interface BlogListEntry {
  slug: string;
  title: string;
  pub_date: string;
}