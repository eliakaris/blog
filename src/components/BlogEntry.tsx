import * as React from 'react';
import { BlogEntryData } from '../../server/BlogEntry';
import './BlogEntry.css';
import 'highlight.js/styles/github.css';

export interface Props {
  blogEntry: BlogEntryData;
}

function BlogEntry({ blogEntry }: Props) {

  const postRoot = 'https://github.com/eliakaris/blog/tree/master/server/data/posts';
  return (
    blogEntry && 
    (
    <article>
      <h1><a href={`/blog/${blogEntry.slug}`} itemProp="url">{blogEntry.title}</a></h1>
      <div className="meta">
        <p>
          <time itemProp="dateCreated" dateTime={`${blogEntry.pub_date}`}>
            {blogEntry.pretty_pub_date}
          </time>
          &nbsp;âˆ™&nbsp;
          <a href={`${postRoot}/${blogEntry.slug}.md`}>
            History
          </a>
        </p>
      </div>
      <div dangerouslySetInnerHTML={{__html: blogEntry.html}} />
    </article>
    )
  );
}

export default BlogEntry;