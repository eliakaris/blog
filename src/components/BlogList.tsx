import * as React from 'react';
import { BlogListEntry } from '../../server/BlogEntry';
import BlogListItem from './BlogListItem';

export interface Props {
  blogEntries: BlogListEntry[];
}

function BlogList({ blogEntries }: Props) {
  var blogItems: JSX.Element[];
  blogItems = [];
  if (blogEntries && blogEntries.length > 0) {
    var years = {};
    blogEntries.map((blogItem) => {
      var year = new Date(blogItem.pub_date).getFullYear();
      if (!years[year]) {
        years[year] = true;
        blogItems.push(<h1 key={year}>{year}</h1>);
      }

      blogItems.push(<BlogListItem blogItem={blogItem} key={blogItem.slug} />);
      blogItems.push(<hr key={blogItem.slug + 'hr'} />);
    });
  }

  return (
    <div className="col-lg-8 col-md-10 mx-auto">
      {blogItems}
    </div>
  );
}

export default BlogList;