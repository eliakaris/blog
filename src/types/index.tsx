// src/types/index.tsx
import { BlogEntryData, BlogListEntry } from '../../server/BlogEntry';

export interface StoreState {
  blogEntry: BlogEntryData;
  blogEntries: BlogListEntry[];
}
