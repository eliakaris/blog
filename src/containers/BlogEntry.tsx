import * as React from 'react';
import { Dispatch } from 'react-redux';
import BlogEntry, { Props } from '../components/BlogEntry';
import { StoreState } from '../types/index';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import { RequestBlogEntryAction } from '../actions/index';

interface BlogEntryContainerProps extends Props {
  match: {
    params: {
      slug: string;
    }
  };
  dispatch: Dispatch<RequestBlogEntryAction>;
}

class BlogEntryContainer extends React.Component<BlogEntryContainerProps> {

  componentDidMount() {
    this.props.dispatch(actions.getBlogEntry(this.props.match.params.slug));
  }

  render() {
    return (
      <BlogEntry {...this.props} />
    );
  }
}

export function mapStateToProps({ blogEntry }: StoreState) {
  return {
    blogEntry
  };
}

export default connect(mapStateToProps)(BlogEntryContainer);