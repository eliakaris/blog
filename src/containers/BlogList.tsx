import * as React from 'react';
import { Dispatch } from 'react-redux';
import BlogList, { Props } from '../components/BlogList';
import { StoreState } from '../types/index';
import { connect } from 'react-redux';
import * as actions from '../actions/index';
import { RequestBlogEntriesAction } from '../actions/index';

interface BlogListContainerProps extends Props {
  dispatch: Dispatch<RequestBlogEntriesAction>;
}

class BlogListContainer extends React.Component<BlogListContainerProps> {

  componentDidMount() {
    this.props.dispatch(actions.getBlogEntries());
  }

  render() {
    return (
      <BlogList {...this.props} />
    );
  }
}

export function mapStateToProps({ blogEntries }: StoreState) {
  return {
    blogEntries
  };
}

export default connect(mapStateToProps)(BlogListContainer);