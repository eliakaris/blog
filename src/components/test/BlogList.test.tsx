import * as React from 'react';
import BlogList from '../BlogList';
import * as ShallowRenderer from 'react-test-renderer/shallow';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import { BlogListEntry } from '../../types/BlogEntry';

configure({ adapter: new ReactSixteenAdapter() });

describe('<BlogList /> tests', () => {
  let renderer: any;

  beforeAll(() => {
    renderer = ShallowRenderer.createRenderer();
  });

  it('renders a snapshot of empty blog list', () => {
    const component = renderer.render(<BlogList blogEntries={[]} />);
    expect(component).toMatchSnapshot();
  });

  it('renders a snapshot of a single entry blog list', () => {
    const blogEntry: BlogListEntry = {
      slug: "test-blog-entry",
      title: "test blog entry",
      pub_date: "20171226",
      pretty_pub_date: "December 26th, 2017",
      summary: "this is a summary text"
    };

    const blogEntries = [
      blogEntry
    ];
    const component = renderer.render(<BlogList blogEntries={blogEntries} />);
    expect(component).toMatchSnapshot();
  });

  it('renders a snapshot of a multiple entry blog list', () => {
    const blogEntry1: BlogListEntry = {
      slug: "test-blog-entry-1",
      title: "test blog entry 1",
      pub_date: "20171226",
      pretty_pub_date: "December 26th, 2017",
      summary: "this is a summary text 1"
    };

    const blogEntry2: BlogListEntry = {
      slug: "test-blog-entry-2",
      title: "test blog entry 2",
      pub_date: "20171225",
      pretty_pub_date: "December 25th, 2017",
      summary: "this is a summary text 2"
    };

    const blogEntries = [
      blogEntry1,
      blogEntry2
    ];
    const component = renderer.render(<BlogList blogEntries={blogEntries} />);
    expect(component).toMatchSnapshot();
  });
});