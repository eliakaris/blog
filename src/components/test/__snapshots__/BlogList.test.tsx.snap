// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogList /> tests renders a snapshot of a multiple entry blog list 1`] = `
<div
  className="col-lg-8 col-md-10 mx-auto"
>
  <h1>
    2018
  </h1>
  <BlogListItem
    blogItem={
      Object {
        "pretty_pub_date": "December 26th, 2017",
        "pub_date": 2018-01-26T08:00:00.000Z,
        "slug": "test-blog-entry-1",
        "summary": "this is a summary text 1",
        "title": "test blog entry 1",
      }
    }
  />
  <hr />
  <BlogListItem
    blogItem={
      Object {
        "pretty_pub_date": "December 25th, 2017",
        "pub_date": 2018-01-25T08:00:00.000Z,
        "slug": "test-blog-entry-2",
        "summary": "this is a summary text 2",
        "title": "test blog entry 2",
      }
    }
  />
  <hr />
</div>
`;

exports[`<BlogList /> tests renders a snapshot of a single entry blog list 1`] = `
<div
  className="col-lg-8 col-md-10 mx-auto"
>
  <h1>
    2018
  </h1>
  <BlogListItem
    blogItem={
      Object {
        "pretty_pub_date": "December 26th, 2017",
        "pub_date": 2018-01-26T08:00:00.000Z,
        "slug": "test-blog-entry",
        "summary": "this is a summary text",
        "title": "test blog entry",
      }
    }
  />
  <hr />
</div>
`;

exports[`<BlogList /> tests renders a snapshot of empty blog list 1`] = `
<div
  className="col-lg-8 col-md-10 mx-auto"
>
  
</div>
`;
